### ISSUE TITLE
False positive with `no-sync`

### ISSUE THREAD

**Issue by fflorent** [eslint/eslint#12601]
**What rule do you want to change?** no-eslint

**Does this change cause the rule to produce more or fewer warnings?** fewer warnings

**How will the change be implemented? (New option, new default behavior, etc.)?** New option

**Please provide some example code that this change will affect:**

```js
const properties = JSON.parse(someInput);
const enableSync = properties.enableSync;
```

**What does the rule currently do for this code?** It produces a false-positive : the above `enableSync` is a boolean property. However, we would like to consider `enableSync` as a property, not a function.

**What will the rule do after it's changed?**

I see two ways to fix the above example:
 - change the parser to detect whether the property is actually called;
 - or add a new option in order to allow exceptions through AST selectors;

What do you think? The second option could be a better approach in order to give more freedom to the users.

**Are you willing to submit a pull request to implement this change?** Yes. 

I already have a patch to propose, but I'd like to discuss about the good approach to fix this first :).

**Comment by kaicataldo** [comment]
Thanks for the issue. Just to clarify, I think you meant `no-sync` for the rule you want to change, correct?

I'm supportive of this change! Can you expand on why you think AST selectors would be a better option than a list of identifiers to ignore?
